### DISCUSSION TITLE
V3 Pagination Query string in url not showing

### DISCUSSION THREAD

**Discussion by ilo-begashvili** [livewire/livewire#6903] [1. Help]
Hello

i have problem about Pagination on livewire v3

i have one component where im dispatching event

in second component i must read event and fire query

i have logic ( i need to be logic) when i receive event data then run query with additional params (by event)

but i have issue about pagination, everything works well except pagination query string in url
pages also works well when i click second page it shows second page data but url not changing, url is without ?page=2

https://github.com/livewire/livewire/assets/17001345/27cd8d2d-a538-4b97-a6c9-3002653f0661

i will share my code

  ```
  #[On('filterProperties')]

    public function getEvent($eventData = null)
    {
        // $this->resetPage();
        if($eventData) $this->filters = $eventData;
    }


    public function getVehicles()
    {
        $query = Vehicle::with(['brand', 'model', 'year', 'location:id,name']);
        $query->WithFilters($this->filters);   

        return $query->limit(20)
            ->orderBy('id', 'desc')
            ->paginate(10);
    }

    public function render()
    {
        if($this->filters){
            return view('livewire.vehicles-container',['vehicles' => $this->getVehicles()]);
        }else{
            return view('livewire.vehicles-container',['vehicles' => null]);
        }
    }
```

------------

in this logic (below), pagination query string is showing in url, but it makes two query
first getting all data without filter and then when filters is received by event, then runs second query with filter

https://github.com/livewire/livewire/assets/17001345/af9a71a9-4c5f-4595-bea3-b31f837f0cff

  ```
   public function render($eventData = null)
    {

        $query = Vehicle::with(['brand', 'model', 'year', 'location:id,name']);
        if($this->filters) $query->WithFilters($this->filters);   

        $this->vehicles = $query->limit(20)
            ->orderBy('id', 'desc')
            ->paginate(10);
        //dd($this->vehicles);
        return view('livewire.vehicles-container', ['vehicles' => $this->vehicles]);
    }
```

    
    and its bad idea for performace and server..

**Comment by PopovicV** [comment]
I'm not 100% sure, but I assume the problem is with $this->paginators array not initialising properly

Had the same problem, and I found this workaround:

```
public function render()
  {
      if($this->filters){
          return view('livewire.vehicles-container',['vehicles' => $this->getVehicles()]);
      }else{
          return view('livewire.vehicles-container',['vehicles' => new LengthAwarePaginator([], 0, 20, $this->getPage())]);
      }
  }
```
